index.js

```diff
@@ -0,0 +1,59 @@
+const bodyParser = require('body-parser')
+const express = require('express')
+
+const teams = require('./teams.json')
+
+const app = express()
+
+app.get('/teams', (request, response) => {
+  response.send(teams)
+})
+
+app.get('/teams/:input', (request, response) => {
+  const requestInfo = request.params.input
+  const teamByParams = teams.filter(team => {
+    return (
+      team.id === Number(requestInfo) ||
+      team.abbreviation.toUpperCase() === requestInfo.toUpperCase()
+    )
+  })
+  return teamByParams.length
+    ? response.send(teamByParams)
+    : response.sendStatus(404)
+})
+app.use(bodyParser.json())
+app.post('/teams', (request, response) => {
+  let {
+    id,
+    location,
+    mascot,
+    abbreviation,
+    conference,
+    division
+  } = request.body
+
+  if (!location || !mascot || !abbreviation || !conference || !division) {
+    response
+      .status(400)
+      .send(
+        'The following are required:  id, location, mascot, abbreviation, conference, division'
+      )
+  }
+  id = Object.keys(teams).length + 1
```

Comment:
Removing `id` from the destructure above means you can use const up there and chang this to 

```JavaScript
const id = Object.keys(teams).length  + 1
```